exporters/trace/ocagent/src/main/java/io/opencensus/exporter/trace/ocagent/OcAgentTraceServiceStreamObservers.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.trace.ocagent;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.grpc.stub.StreamObserver;
+import io.opencensus.proto.agent.trace.v1.CurrentLibraryConfig;
+import io.opencensus.proto.agent.trace.v1.ExportTraceServiceRequest;
+import io.opencensus.proto.agent.trace.v1.ExportTraceServiceResponse;
+import io.opencensus.proto.agent.trace.v1.UpdatedLibraryConfig;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.config.TraceConfig;
+import io.opencensus.trace.config.TraceParams;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/*>>>
+import org.checkerframework.checker.nullness.qual.Nullable;
+*/
+
+/** Stream observers of OC-Agent Trace Service. */
+final class OcAgentTraceServiceStreamObservers {
+
+  private OcAgentTraceServiceStreamObservers() {}
+
+  private static final Logger logger =
+      Logger.getLogger(OcAgentTraceServiceStreamObservers.class.getName());
+
+  /*
+   * gRPC-Java has good abstractions over the transport layer, and there's no way we can know the
```

Comment:
Why do we not use this:
https://grpc.io/grpc-java/javadoc/io/grpc/ManagedChannel.html#getState-boolean-?

This seems to achieve what you need to check if the agent is up and accepts connections. If this works I think we simplify a lot the design.

Reply:
I see, I didn't realize we can check channel connection with this method. This will be convenient for the worker thread that tries to connect to Agent, though I think the logic for `StreamObserver`s should stay the same.